### PR TITLE
Update securesubmit.php

### PR DESCRIPTION
**Pull request by ttnnkkrr**

### CODE REVIEW

File: catalog/controller/extension/payment/securesubmit.php [hunk 1]
```diff
@@ -461,6 +461,12 @@ private function sessionVar($key = null, $val = null)
...
         //unset($this->session->data[$HPS_KEY]);
+        if (!is_array($this->session->data)){
+            $this->session->data = array();
+        }
+         if (!key_exists($HPS_KEY,$this->session->data) || !is_array($this->session->data[$HPS_KEY])){
+             $this->session->data[$HPS_KEY] = array();
+         }
         if (!key_exists('lastErrorTime',$this->session->data[$HPS_KEY])){
             $this->session->data[$HPS_KEY]['lastErrorTime'] = 0;
         }
```
Comment (SenthilKumar-86) [inline catalog/controller/extension/payment/securesubmit.php:475]: is it possible to replace these three if conditions with this one line code, like 
$this->session->data[$HPS_KEY]['lastErrorTime'] = (!empty($this->session->data[$HPS_KEY]['lastErrorTime'])) ? $this->session->data[$HPS_KEY]['lastErrorTime'] : 0;

File: catalog/controller/extension/payment/securesubmit.php [hunk 2]
```diff
@@ -461,6 +461,12 @@ private function sessionVar($key = null, $val = null)
          */
         $HPS_KEY = $this->securesubmit_fraud_fail_var;
```
Comment (SenthilKumar-86) [inline catalog/controller/extension/payment/securesubmit.php]: shall we add a condition here to make sure $this->securesubmit_fraud_fail_var is not empty?

**Review by SenthilKumar-86 — CHANGES_REQUESTED** [review]

File: catalog/controller/extension/payment/securesubmit.php (hunk 1 above)
Comment (ttnnkkrr) [reply catalog/controller/extension/payment/securesubmit.php:475]: @SenthilKumar-86 I suppose we could but dont you think this way is easier to read?

**Comment by ttnnkkrr** [comment]
@SenthilKumar-86 on $this->securesubmit_fraud_fail_var good catch

**Review by SenthilKumar-86 — APPROVED** [review]

File: catalog/controller/extension/payment/securesubmit.php (hunk 1 above)
Comment (SenthilKumar-86) [reply catalog/controller/extension/payment/securesubmit.php:475]: Yes i agree readability wise it's easier..

**Review by SenthilKumar-86 — APPROVED** [review]

**Comment by SenthilKumar-86** [comment]
Functionality wise, When we buy product,  orders listed as canceled status in admin side. Can you please check it.

**Review by SenthilKumar-86 — CHANGES_REQUESTED** [review]
Functionality needs to be checked

**Review by SenthilKumar-86 — APPROVED** [review]

**Pull request merged**